File: src/main/java/com/jacuum/engine/SessionTile.java
package com.jacuum.engine;

import com.jacuum.algo.Direction;
import com.jacuum.algo.Tile;
import java.util.Set;

final class SessionTile implements Tile {
    private final int x;
    private final int y;
    private final com.jacuum.map.GameMap map;
    private final Set<String> cleaned;

    SessionTile(int x, int y, com.jacuum.map.GameMap map, Set<String> cleaned) {
        this.x       = x;
        this.y       = y;
        this.map     = map;
        this.cleaned = cleaned;
    }

    @Override public int x()                          { return x; }
    @Override public int y()                          { return y; }
    @Override public boolean isClean()                { return cleaned.contains(x + "," + y); }
    @Override public boolean hasWall(Direction dir)   { return map.hasWall(x, y, dir); }
}

File: src/main/kotlin/com/jacuum/engine/SessionTile.kt
package com.jacuum.engine

import com.jacuum.algo.Direction
import com.jacuum.algo.Tile

internal class SessionTile(
    private val x: Int,
    private val y: Int,
    private val map: com.jacuum.map.GameMap,
    private val cleaned: Set<String>
) : Tile {
    override fun x(): Int = x
    override fun y(): Int = y
    override fun isClean(): Boolean = cleaned.contains("$x,$y")
    override fun hasWall(direction: Direction): Boolean = map.hasWall(x, y, direction)
}
